src/sentry/rules/history/preview.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+from datetime import timedelta
+
+from django.utils import timezone
+
+from sentry.models import Project
+from sentry.rules import rules
+from sentry.rules.history.base import TimeSeriesValue
+from sentry.utils.types import Any, Dict, Sequence
+
+PREVIEW_TIME_RANGE = timedelta(weeks=2)
+
+
+def preview(
+    project: Project,
+    conditions: Sequence[Dict[str, Any]],
+    filters: Sequence[Dict[str, Any]],
+    condition_match: str,
+    filter_match: str,
+    frequency_minutes: int,
+) -> Sequence[TimeSeriesValue]:
+    end = timezone.now()
+    start = end - PREVIEW_TIME_RANGE
+    hours = get_hourly_bucket(PREVIEW_TIME_RANGE)
+    hourly_buckets = [0] * hours
+
+    # must have at least one condition to filter activity
+    if len(conditions) == 0:
+        return None
+    # all the currently supported conditions are mutually exclusive
+    elif len(conditions) > 1 and condition_match == "all":
+        return [TimeSeriesValue(start + timedelta(hours=i), 0) for i in range(hours)]
```

Comment:
so this is just true for anything with multiple conditions? not super familiar with general rule logic

Reply:
This should be true for `FirstSeenEventCondition`, `ReappearedEventCondition` and `RegressionEventCondition` which are the ones we have plans to support right now. The other ones aren't answerable using just `Activity` anyways, so we're probably gonna be done another way

Reply:
maybe put a comment in like `represent an invalid condition set that will never fire`